lib/wibox/container/radialprogressbar.lua

```diff
@@ -0,0 +1,264 @@
+    cr:set_line_width(border_width)
+    cr:stroke()
+
+end
+
+-- Set the clip
+function radialprogressbar:before_draw_children(_, cr, width, height)
+    cr:save()
+    local wa = content_workarea(self, width, height)
+    cr:translate(wa.x, wa.y)
+    shape.rounded_bar(cr, wa.width, wa.height)
+    cr:clip()
+    cr:translate(-wa.x, -wa.y)
+end
+
+-- Layout this layout
+function radialprogressbar:layout(_, width, height)
+    if self._private.widget then
+        local wa = content_workarea(self, width, height)
+
+        return { base.place_widget_at(
+            self._private.widget, wa.x, wa.y, wa.width, wa.height
+        ) }
+    end
+end
+
+-- Fit this layout into the given area
+function radialprogressbar:fit(context, width, height)
+    if self._private.widget then
+        local wa = content_workarea(self, width, height)
+        return base.fit_widget(self, context, self._private.widget, wa.width, wa.height)
+    end
+
+    return width, height
+end
+
+--- The widget to wrap in a radial proggressbar.
+-- @property widget
+-- @tparam widget widget The widget
+
+function radialprogressbar:set_widget(widget)
+    if widget then
+        base.check_widget(widget)
+    end
+    self._private.widget = widget
+    self:emit_signal("widget::layout_changed")
+end
+
+--- Get the number of children element
+-- @treturn table The children
+function radialprogressbar:get_children()
+    return {self._private.widget}
+end
+
+--- Replace the layout children
+-- This layout only accept one children, all others will be ignored
+-- @tparam table children A table composed of valid widgets
+function radialprogressbar:set_children(children)
+    self._private.widget = children and children[1]
+    self:emit_signal("widget::layout_changed")
+end
+
+--- Reset this layout. The widget will be removed and the rotation reset.
+function radialprogressbar:reset()
+    self:set_widget(nil)
+end
+
+for k,v in ipairs {"left", "right", "top", "bottom"} do
+    radialprogressbar["set_"..v.."_padding"] = function(self, val)
+        self._private.paddings = self._private.paddings or {}
+        self._private.paddings[v] = val
+        self:emit_signal("widget::redraw_needed")
+    end
+end
+
+--- The padding between the outline and the progressbar.
+--@DOC_wibox_container_radialprogressbar_padding_EXAMPLE@
+-- @property paddings
+-- @tparam[opt=0] table|number paddings A number or a table
+-- @tparam[opt=0] number paddings.top
+-- @tparam[opt=0] number paddings.bottom
+-- @tparam[opt=0] number paddings.left
+-- @tparam[opt=0] number paddings.right
+
+--- The progressbar value.
+--@DOC_wibox_container_radialprogressbar_value_EXAMPLE@
+-- @property value
+-- @tparam number value Between min_value and max_value
+
+function radialprogressbar:set_value(val)
+    if not val then self._percent = 0; return end
+
+    if val > self._private.max_value then
+        self:set_max_value(val)
+    elseif val < self._private.min_value then
+        self:set_min_value(val)
```

Comment:
Does this work? As far as I can tell, there is no `max/min`, but `maximum/minimum`.